ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py

```diff
@@ -2482,3 +2482,44 @@ def test_isreal(
         fn_name="isreal",
         x=x[0],
     )
+
+
+# gcd
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        shared_dtype=True,
+        min_num_dims=1,
+        max_num_dims=3,
+        min_value=-100,
+        max_value=100,
+        allow_nan=False,
+    ),
+    num_positional_args=helpers.num_positional_args(fn_name="diff"),
+)
+def test_diff(
+    dtype_and_x,
+    with_out,
```

Comment:
Need to define strategies for `with_out`, `as_variable` etc in the `@given` decorator

Reply:
Hi, @simonetgordon added strategies in <code>@given</code> decoder. Please check if it's correct.
Thanks!